__tests__/PropTypesDevelopmentReact15.js

```diff
@@ -875,6 +875,89 @@ describe('PropTypesDevelopmentReact15', () => {
     });
   });
 
+  describe('ElementWithType Types', () => {
+    it('should fail for mismatched component.type', () => {
+      typeCheckFail(
+        PropTypes.elementWithType('div'),
+        <span/>,
+        "Warning: Failed prop type: Invalid prop `testProp` of type [span] supplied to `testComponent`, expected one of type `div`."
+      );
+    });
+
+    it('should pass for matched component.type', () => {
+      typeCheckPass(PropTypes.elementWithType('div'), <div />);
+    });
+
+    it('should pass if no component given', () => {
+      typeCheckPass(PropTypes.elementWithType('div'));
+    });
+
+    describe('factory error', () => {
+      it('should fail if using any value thats not a function nor a string for expectedTypes', () => {
+        typeCheckFail(
+          PropTypes.elementWithType(1),
+          <span />,
+          "Warning: Failed prop type: Property `testProp` of component `testComponent` has invalid PropType notation inside elementWithType."
+        );
+      });
+    });
+
+    describe('custom component', () => {
+      const MyComponent = () => <div></div>;
+      var myComponentPropType;
+      beforeEach(() => {
+        myComponentPropType = PropTypes.elementWithType(MyComponent)
+      });
+
+      it('should fail for mismatched component.type', () => {
+        typeCheckFail(
+          myComponentPropType,
+          <span />,
+          "Warning: Failed prop type: Invalid prop `testProp` of type [span] supplied to `testComponent`, expected one of type `MyComponent`."
+        );
+      });
+
+      it('should pass for matched component.type', () => {
+        typeCheckPass(myComponentPropType, <MyComponent />);
+      });
+    });
+
+    describe('given component mismatch', () => {
+      it('should warn if given component.type other than function or array of functions', () => {
```

Comment:
Array language can be removed here (also in the docs)

Reply:
going to change to `should warn if given invalid component`